jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -214,6 +215,15 @@ def _is_unique(m):
         if subset_type == 'spatial':
             layer_state.linewidth = 1
 
+            # change opacity for live-collapsed spectra from spatial subsets in Cubeviz:
+            layers_and_marks = zip(
+                self.state.layers, self._get_marks_for_layers(self.state.layers)
+            )
```

Comment:
is there a reason this needs to loop over all layers/marks?  Would acting directly _only_ on `layer_state` (and the matching mark(s)) sent to this method get overwritten after-the-fact?

Reply:
I think they'll be linked anyways via the plot options (unless we want to break that link specifically for opacities... we definitely want the colors to remain linked) 🤔

Reply:
This (in retrospect, obvious) improvement borrows from the code in the other `if`-block in this method: 20ad4d1. It now loops over only the spatial subsets.